metagraph/tests/graph/all/test_dbg_helpers.cpp

```diff
@@ -126,6 +132,62 @@ build_graph<DBGBitmap>(uint64_t k,
     return graph;
 }
 
+void writeFastaFile(const std::vector<std::string>& sequences, const std::string& outputFilename) {
+    std::ofstream fastaFile(outputFilename);
+
+    if (!fastaFile.is_open()) {
+        std::cerr << "Error: Unable to open the output file." << std::endl;
+        return;
+    }
+
+    for (size_t i = 0; i < sequences.size(); ++i) {
+        fastaFile << ">"<< "\n" << sequences[i] << "\n";
+    }
+
+    fastaFile.close();
+}
+template <>
+std::shared_ptr<DeBruijnGraph>
+build_graph<DBGSSHash>(uint64_t k,
+                       std::vector<std::string> sequences,
+                       DeBruijnGraph::Mode mode) {
+    if (sequences.empty())
+        return std::make_shared<DBGSSHash>(k, mode);
+
+    // use DBGHashString to get contigs for SSHash
+    auto string_graph = build_graph<DBGHashString>(k, sequences, mode);
+
+    std::vector<std::string> contigs;
+    size_t num_kmers = 0;
+    string_graph->call_sequences([&](const std::string &contig, const auto &path) {
+        contigs.push_back(contig);
+        num_kmers += path.size();
+    }, 1, mode != DeBruijnGraph::BASIC);
+
+    if (contigs.empty())
+        return std::make_shared<DBGSSHash>(k, mode);
+
+    std::string dump_path = "../tests/data/sshash_sequences/contigs.fa";
+    writeFastaFile(contigs, dump_path);
+
+    std::shared_ptr<DBGSSHash> graph;
+    try {
+        graph = std::make_shared<DBGSSHash>(dump_path, k, mode);
+    } catch (const std::runtime_error &e) {
+        if (strcmp(e.what(), "each partition must contain more than one key: use less partitions") == 0) {
+            return build_graph<DBGHashOrdered>(k, sequences, mode);
```

Comment:
Why?

Reply:
The idea here is that pthash assumes that each partition has at least two elements, which is violated for small graphs. So, we first try to run with DBGSSHash and switch over to a different graph if it fails.

See here for the issue I opened with pthash about this"
https://github.com/jermp/pthash/issues/63

The ideal thing would be to modify pthash to work with empty or single-element partitions so we don't have to do this hack.

Reply:
But it doesn't test DBGSSHash if it builds some other kind of graph, so it would make the test somewhat bogus on DBGSSHash... Okay, I suppose we'll have to look into pthash eventually and try to fix it ourselves, if small cases are of no interest to Giulio. Maybe not necessarily as a part of this PR, but then we should keep track of it in an issue or smt.

Reply:
Should be fixed now that jermp/pthash#63 is settled. Closing for now, will reopen if tests still fail.